src/keymap-manager.coffee

```diff
@@ -474,36 +507,28 @@ class KeymapManager
     # keystroke.
     if partialMatches.length > 0
       event.preventDefault()
-      enableTimeout = (
-        @pendingStateTimeoutHandle? or
-        foundMatch or
-        characterForKeyboardEvent(@queuedKeyboardEvents[0])?
-      )
-
-      @enterPendingState(partialMatches, enableTimeout)
       @emitter.emit 'did-partially-match-binding', {
         keystrokes,
         eventType: event.type,
         partiallyMatchedBindings: partialMatches,
         keyboardEventTarget: target
       }
-    else
+    else if exactMatch is null
       @emitter.emit 'did-fail-to-match-binding', {
         keystrokes,
         eventType: event.type,
         keyboardEventTarget: target
       }
-      if @pendingPartialMatches?
-        @terminatePendingState()
-      else
-        # Some of the queued keyboard events might have inserted characters had
-        # we not prevented their default action. If we're replaying a keystroke
-        # whose default action was prevented and no binding is matched, we'll
-        # simulate the text input event that was previously prevented to insert
-        # the missing characters.
-        @simulateTextInput(event) if event.defaultPrevented
-
-        @clearQueuedKeystrokes()
+      # Some of the queued keyboard events might have inserted characters had
+      # we not prevented their default action. If we're replaying a keystroke
+      # whose default action was prevented and no binding is matched, we'll
+      # simulate the text input event that was previously prevented to insert
+      # the missing characters.
+      @simulateTextInput(event) if event.defaultPrevented and event.type is 'keydown'
```

Comment:
Could this piece be moved into the outer method as well so that this method can evolve into something with zero side effects?